include/wlr/types/wlr_input_method_v2.h

```diff
@@ -41,16 +41,34 @@ struct wlr_input_method_v2 {
 	bool client_active; // state known to the client
 	uint32_t current_serial; // received in last commit call
 
+	struct wlr_input_method_keyboard_grab_v2 *keyboard_grab;
+
 	struct wl_list link;
 
 	struct wl_listener seat_destroy;
 
 	struct {
 		struct wl_signal commit; // (struct wlr_input_method_v2*)
+		struct wl_signal keyboard_grab; // (struct wlr_input_method_v2*)
 		struct wl_signal destroy; // (struct wlr_input_method_v2*)
 	} events;
 };
 
+struct wlr_input_method_keyboard_grab_v2 {
+	struct wl_resource *resource;
+	struct wlr_input_method_v2 *input_method;
+	struct wlr_seat_client *seat_client;
+	struct wlr_keyboard *keyboard;
+
+	struct wl_listener keymap_listener;
```

Comment:
So far the naming convention for listeners has been `<object>_<event>`. For consistency we should rename those to `keyboard_keymap`, `keyboard_repeat_info` and `keyboard_destroy`.

Reply:
Fixed.